pkg/backend/watch.go

```diff
@@ -108,8 +99,52 @@ func Watch(ctx context.Context, b Backend, stack Stack, op UpdateOperation,
 			display.PrintfWithWatchPrefix(time.Now(), "",
 				op.Opts.Display.Color.Colorize(colors.SpecImportant+"Update complete."+colors.Reset+"\n"))
 		}
-
 	}
 
 	return nil
 }
+
+func watchPaths(root string, paths []string) (chan string, func(), result.Result) {
+	args := []string{"--origin", root}
+	for _, p := range paths {
+
+		watchPath := ""
+		if path.IsAbs(p) {
+			watchPath = p
+		} else {
+			watchPath = path.Join(root, p)
+		}
+
+		args = append(args, "--watch", watchPath)
+	}
+
+	watchCmd, err := exec.LookPath("pulumi-watch")
```

Comment:
So
A) We should probably envvar this so people can override it PULUMI_WATCH_BINARY?
B) We should support the "watch.exe" being next to "pulumi.exe" but not on path.

Reply:
Does `PULUMI_WATCH_BINARY` make sense if we require specific command line args?

Finding side-by-side `pulumi-watch` makes sense as a fallback, in case the user is invoking Pulumi by installing it to, e.g.: /opt/pulumi and invoking it via `/opt/pulumi up`. Will add that.

Reply:
I've refactored out the `exec.LookPath` to support finding the .exe on Windows & look for a side by side binary. After installing `pulumi`, `pulumi-watch`, and `pulumi-language-yaml` to ~/.pulumi-dev/bin, I was able to run:

```
$ PATH="" ~/.pulumi-dev/bin/pulumi watch
Watching (dev):
10:53:09.646[                    ] Updating...
10:53:10.624[                    ] Update complete.
10:53:15.029[                    ] Updating...
10:53:15.566[                 bar] create random:index/randomUuid:RandomUuid
10:53:15.664[                 bar] done create random:index/randomUuid:RandomUuid
10:53:15.865[                    ] Update complete.
```

If I removed either of the YAML or watch binary, it would fail. Sans watch:

```
$ PATH="" ~/.pulumi-dev/bin/pulumi watch
error: Could not locate pulumi-watch binary
```

Sans YAML:

```
$ PATH="" ~/.pulumi-dev/bin/pulumi watch --debug -v9 --logflow
Watching (dev):
10:57:23.752[                    ] Updating...
10:57:24.117[                    ] Update failed.
```

Not a great UX, but I think that's nothing new.